### ISSUE TITLE
AssertionError: passing these results is not yet implemented

### ISSUE THREAD

**Issue by jhoenicke** [ultimate-pa/ultimate#218]
The following example program throws the error mentioned in the title (using web interface).  The problem seems to be the define of OLED_BUFSIZE  as 128*64/8.  When replacing it with 1024 the program goes through.

```c
typedef unsigned char uint8_t;

#define OLED_WIDTH   128
#define OLED_HEIGHT  64
#define OLED_BUFSIZE (OLED_WIDTH * OLED_HEIGHT / 8)

static uint8_t _oledbuffer[OLED_BUFSIZE];


#define OLED_OFFSET(x, y) (OLED_BUFSIZE - 1 - (x) - ((y)/8)*OLED_WIDTH)
#define OLED_MASK(x, y) (1 << (7 - (y) % 8))

#define max(X,Y) ((X) > (Y) ? (X) : (Y))
#define min(X,Y) ((X) < (Y) ? (X) : (Y))

/*
 * Inverts pixel at x, y
 */
void oledInvertPixel(int x, int y)
{
        if ((x < 0) || (y < 0) || (x >= OLED_WIDTH) || (y >= OLED_HEIGHT)) {
                return;
        }
        _oledbuffer[OLED_OFFSET(x, y)] ^= OLED_MASK(x, y);
}
/*
 * Inverts box between (x1,y1) and (x2,y2) inclusive.
 */
void oledInvert(int x1, int y1, int x2, int y2)
{
        x1 = max(x1, 0);
        y1 = max(y1, 0);
        x2 = min(x2, OLED_WIDTH - 1);
        y2 = min(y2, OLED_HEIGHT - 1);
        for (int x = x1; x <= x2; x++) {
                for (int y = y1; y <= y2; y++) {
                        oledInvertPixel(x,y);
                }
        }
}

int getInt();

int main()
{
    oledInvert(getInt(), getInt(), getInt(), getInt());
}
```

**Comment by alexandernutz** [comment]
Other witnesses for this error:
memsafety/960521-1_false-valid-deref.i
memsafety/960521-1_false-valid-free.i

**Comment by alexandernutz** [comment]
btw. the problem here is that an array access position is given with an expression that has side effects. Which we do not expect/support. Here, the side effect is a division-by-zero check.

**Comment by Heizmann** [comment]
Workaroud: switch off division by zero check in preferences

**Comment by Heizmann** [comment]
Although there is this workaround, we want to have to fix this (not urgent).
Difficulty: if the check is in the initialization of variables, the corresponding assert has to be added to the `ULTIMATE.init` procedure.

**Comment by Heizmann** [comment]
Although there is this workaround, we want to have to fix this (not urgent).
Difficulty: if the check is in the initialization of variables, the corresponding assert has to be added to the `ULTIMATE.init` procedure.

**Comment by schuessf** [comment]
The mentioned program works now. This issue was fixed by 1c51228.
